File: formula-java/src/main/java/org/formula/StaticResolvable.java
package org.formula;

import java.util.Optional;
import org.formula.context.DataContext;

public class StaticResolvable implements Resolvable {
    private final ResolvedValue resolved;

    public static Resolvable of(ResolvedValue value) {
        return new StaticResolvable(value);
    }

    @Override
    public ResolvedValue resolve() {
        return Optional.ofNullable(resolved).orElse(ResolvedValue.none());
    }

    @Override
    public ResolvedValue resolve(DataContext context) {
        return resolve();
    }

    private StaticResolvable(ResolvedValue resolved) {
        this.resolved = resolved;
    }
}

File: py/formula-py/formula/static_resolvable.py
from formula import Resolvable, ResolvedValue
from formula.resolved_value import resolved_value


class StaticResolvable(Resolvable):

    def __init__(self, value: str | int | float | bool | ResolvedValue):
        self.value = resolved_value(value)

    def resolve(self, context=None):
        return self.value

    def as_formula(self):
        return self.value.as_text()

